src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_util.py

```diff
@@ -0,0 +1,195 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import register_cli_argument, cli_command
+from azure.cli.core.commands.parameters import ignore_type
+from azure.cli.core.commands.arm import cli_generic_update_command
+
+
+# MANAGEMENT CLIENT FACTORIES
+def get_monitor_management_client(_):
+    from azure.cli.command_modules.monitor.sdk import InsightsManagementClient
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    return get_mgmt_service_client(InsightsManagementClient)
+
+
+def get_monitor_autoscale_settings_operation(kwargs):
+    return get_monitor_management_client(kwargs).autoscale_settings
+
+
+def get_monitor_service_diagnostic_settings_operation(kwargs):
+    return get_monitor_management_client(kwargs).service_diagnostic_settings
+
+
+def get_monitor_alert_rules_operation(kwargs):
+    return get_monitor_management_client(kwargs).alert_rules
+
+
+def get_monitor_alert_rule_incidents_operation(kwargs):
+    return get_monitor_management_client(kwargs).alert_rule_incidents
+
+
+def get_monitor_log_profiles_operation(kwargs):
+    return get_monitor_management_client(kwargs).log_profiles
+
+
+# DATA CLIENT FACTORIES
+def get_monitor_client(_):
+    from azure.monitor import InsightsClient
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    return get_mgmt_service_client(InsightsClient)
+
+
+def get_monitor_usage_metrics_operation(kwargs):
+    return get_monitor_client(kwargs).usage_metrics
+
+
+def get_monitor_event_categories_operation(kwargs):
+    return get_monitor_client(kwargs).event_categories
+
+
+def get_monitor_events_operation(kwargs):
+    return get_monitor_client(kwargs).events
+
+
+def get_monitor_tenant_events_operation(kwargs):
+    return get_monitor_client(kwargs).tenant_events
+
+
+def get_monitor_metric_definitions_operation(kwargs):
+    return get_monitor_client(kwargs).metric_definitions
+
+
+def get_monitor_metrics_operation(kwargs):
+    return get_monitor_client(kwargs).metrics
+
+
+# COMMANDS UTILITIES
+
+def create_service_adapter(service_model, service_class):
+    def _service_adapter(method_name):
+        return '{}#{}.{}'.format(service_model, service_class, method_name)
+
+    return _service_adapter
+
+
+# pylint: disable=too-few-public-methods
+class ServiceGroup(object):
+    def __init__(self, scope, client_factory, service_adapter=None):
+        self._scope = scope
+        self._factory = client_factory
+        self._service_adapter = service_adapter or (lambda name: name)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def group(self, group_name):
+        return CommandGroup(self._scope, group_name, self._factory, self._service_adapter)
+
+
+class CommandGroup(object):
```

Comment:
This is duplicate with the sql command module. Consider move it up to the `azure-cli-core` for reuse.